week2/hw2/memberSerch.js

```diff
@@ -0,0 +1,181 @@
+data.forEach((member) => {
+    const row = document.createElement("tr");
+    row.setAttribute("id", member.id)
+
+    row.innerHTML = `
+      <td><input type="checkbox" class="check" /></td>
+      <td>${member.name}</td>
+      <td>${member.englishName}</td>
+      <td><a href="https://github.com/${member.github}" target="_blank">${member.github}</a></td>
+      <td>${member.gender}</td>
+      <td>${member.role}</td>
+      <td>${member.firstWeekGroup}</td>
+      <td>${member.secondWeekGroup}</td>
+    `;
+
+    tbody.appendChild(row);
+  });
+}
+
+searchBtn.addEventListener("click", () => {
+  const searchName = document.querySelector(".name").value;
+  const searchEngName = document.querySelector(".eng_name").value.toLowerCase();
+  const searchGithub = document.querySelector(".git_id").value.toLowerCase();
+  const searchGender = document.querySelector('select#gender').value;
+  const searchRole = document.querySelector('select#role').value;
+  const searchWeek1 = document.querySelector(".week1").value;
+  const searchWeek2 = document.querySelector(".week2").value;
+
+  const filteredData = getData.filter(member => 
+    member.name.includes(searchName) && 
+    member.englishName.toLowerCase().includes(searchEngName) &&
+    member.github.toLowerCase().includes(searchGithub) &&
+    (searchGender === "none" || member.gender === searchGender) &&
+    (searchRole === "none" || member.role.toLowerCase() === searchRole) &&
+    (searchWeek1 === "" || member.firstWeekGroup == searchWeek1) &&
+    (searchWeek2 === "" || member.secondWeekGroup == searchWeek2)
+  );
+
+  tbody.innerHTML = "";
+  addTable(filteredData);
+});
+
+resetBtn.addEventListener("click", () => {
+  // document.querySelector(".name").value = '';
+  // document.querySelector(".eng_name").value = '';
+  // document.querySelector(".git_id").value = '';
+  // document.querySelector('select#gender').value = '';
+  // document.querySelector('select#role').value = '';
+  // document.querySelector(".week1").value = '';
+  // document.querySelector(".week2").value = '';
+
+  window.location.reload();
+});
+
+delBtn.addEventListener("click", () => {
+  const checked = document.querySelectorAll(".check:checked");
+  const checkedTrs = Array.from(checked).map((checkedTr) => parseInt(
+    checkedTr.closest("tr").id
+  ))
+  const newMember = getData.filter(member => {
+    return !checkedTrs.includes(member.id);
+});
+  console.log(newMember);
+
+  localStorage.setItem("membersData", JSON.stringify(newMember))
+
+  // window.location.reload();
+  tbody.innerHTML = "";
+  addTable(newMember);
+});
+
+
+
+selectAll.addEventListener("change", () => {
+  const isChecked = selectAll.checked;
+
+  const dataCheckbox = tbody.querySelectorAll(".check")
+  dataCheckbox.forEach(checkbox => {
+    checkbox.checked = isChecked;
+  });
+});
+
+showBtn.addEventListener("click", () => {
+  modal.showModal();
+});
+
+closeModalBtn.addEventListener("click", () => {
+  modal.close();
+});
+
```

Comment:
```suggestion
// 모달 열기 및 닫기 함수 분리 (화살표 함수로 변환)
const openModal = () => modal.showModal();
const closeModal = () => modal.close();

// 이벤트 리스너에서 함수 호출
showBtn.addEventListener("click", openModal);
closeModalBtn.addEventListener("click", closeModal);

```

이렇게 이벤트 핸들러 함수를 별도 함수로 분리하는 것도 좋은 방식이라고 생각합니다!

이렇게 하면, 코드의 가독성과 재사용성이 좋아진다는 장점이 있습니다!

이는 어떻게보다 "무엇을"에 집중하는 선언형 프로그래밍의 개념을 반영한 방식이라고 할 수 있는데, "선언형 프로그래밍" "함수형 프로그래밍" 등에 대해서도 한번 찾아보시면 좋을 것 같습니다!